### PR TITLE
Add Affirm and Afterpay providers to additional payment method setup screen

### LINKED ISSUE #6274: [BNPL][MVP] Enable Buy Now, Pay Later (BNPL) payment methods for eligible WooPayments merchants
### _Scenario 1: Choosing from available providers._

**As a WooPayments merchant eligible for BNPL,**
_When I visit the Enable Additional payment methods screen,_
_I want to see the available BNPL providers in addition to the other Alternative Payment methods and the fees they apply, so I can make an informed decision to select the ones I want to activate._

### Acceptance criteria

1. The current screen is divided in two sections: "Popular in your country" and "Buy Now, Pay Later"
2. Display available BNPL providers and their associated fees on its own section. Make sure that country restrictions apply.
3. Display available APM providers and their associated fees on its own section. Make sure that country restrictions apply.
4. Allow merchants to select the provider(s) they want to activate.
5. Upon selecting and activating the provider(s), merchants should receive confirmation of the activated provider(s).

### Designs

Screenshot 2023-05-11 at 20 53 47

Screenshot 2023-05-11 at 20 54 05

### _Scenario 2: Confirmation step_

**As a WooPayments merchant eligible for BNPL,**
_After viewing the activated BNPL methods,_
_I want to be able to navigate to the WooCommerce home or the WooPayments settings page._

### Acceptance criteria
1. Provide clear navigation options for merchants to return to the WooCommerce home or access the WooPayments settings page after activating the BNPL methods.

Screenshot 2023-05-11 at 20 54 38

### PR DESCRIPTION
**Pull request by KarlisJ**
Fixes #6274

#### Changes proposed in this Pull Request

- Updates the design of additional payment method setup screen to account both BNPL eligible and non-eligible merchants
- Splits the list of payment methods into two blocks 
   - Popular in your country
   - Buy Now, Pay Later

#### Testing instructions
1. [Setup server](https://github.com/Automattic/woocommerce-payments-server/pull/3405)
2. Disable additional payment methods 
image
3. Click Enable to get to the screen at question
4. Observe that Affirm and Afterpay are grouped separately 
image
5. Select both BNPL methods and any local method, click continue and observe that they are properly selected
image
6. Click "Go to payment settings" and observe that the same methods are enabled
7. Disable feature flag in the server (refer to 1st point)
8. Repeat step 2-3 
9. Observe that the screen doesn't have BNPL section

-------------------

- [ ] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

File: includes/class-wc-payments-features.php [hunk 1]
```diff
@@ -276,6 +276,23 @@ public static function is_fraud_protection_welcome_tour_dismissed(): bool {
...
 	}
 
+	/**
+	 * Checks whether Simplify Deposits UI is enabled. Enabled by default.
+	 *
+	 * @return bool
+	 */
+	public static function is_simplify_deposits_ui_enabled(): bool {
+		return '1' === get_option( self::SIMPLIFY_DEPOSITS_UI_FLAG_NAME, '1' );
+	}
```
Comment (KarlisJ) [inline includes/class-wc-payments-features.php]: There is so merge conflict issue that's being resolved. Please ignore this block

File: includes/class-wc-payments-features.php [hunk 2]
```diff
@@ -276,6 +276,23 @@ public static function is_fraud_protection_welcome_tour_dismissed(): bool {
...
+
+	/**
+	 * Checks whether the BNPL Affirm Afterpay is enabled.
+	 */
+	public static function is_bnpl_affirm_afterpay_enabled(): bool {
+		$account = WC_Payments::get_account_service()->get_cached_account_data();
+		return ! isset( $account['is_bnpl_affirm_afterpay_enabled'] ) || true === $account['is_bnpl_affirm_afterpay_enabled'];
+	}
+
```
Comment (KarlisJ) [inline includes/class-wc-payments-features.php]: This block as well as other related work is already merged in the base branch but for some reason GH is showing it as diff.

File: tests/unit/test-class-wc-payments-features.php [hunk 3]
```diff
@@ -390,6 +390,51 @@ public function test_is_frt_review_feature_active_returns_false_when_flag_is_not
...
+		$account_service_mock
+			->expects( $this->once() )
+			->method( 'get_cached_account_data' )
+			->willReturn( [ 'is_bnpl_affirm_afterpay_enabled' => false ] );
+
+		WC_Payments::set_account_service( $account_service_mock );
+
+		$this->assertFalse( WC_Payments_Features::is_bnpl_affirm_afterpay_enabled() );
+	}
+
```
Comment (KarlisJ) [inline tests/unit/test-class-wc-payments-features.php]: Part of base branch, GH for some reason is showing up as diff.

File: changelog/add-6034-feature-flag-bnlp [hunk 4]
```diff
@@ -0,0 +1,5 @@
+Significance: patch
+Type: dev
+Comment: This merge is a feature flog utilities that will be made available for masses only in later stages of the project
+
```
Comment (KarlisJ) [inline changelog/add-6034-feature-flag-bnlp]: Part of base branch, GH for some reason is showing up as diff.

Comment (mgascam) [reply changelog/add-6034-feature-flag-bnlp]: Maybe the latest changes in develop need to be merged in base branch?

Also, I guess we'll need a changelog for this change. Please note that we are introducing changes in the layout, even if the feature flag is not active. I guess we'll need to create the changelog with this in mind, thinking in the `What's new` post.

File: client/additional-methods-setup/upe-preview-methods-selector/test/enable-upe-preview-task.test.js
Comment (mgascam) [inline client/additional-methods-setup/upe-preview-methods-selector/test/enable-upe-preview-task.test.js:1]: Do you plan to add a new test to replace this one?

**Review by mgascam — COMMENTED** [review]
I followed testing instructions and work as expected. Code is also looking good. Just left a couple of minor comments for your consideration.

Also note that there is a conflict, maybe it is a good oportunity to be sure that base branch is up to date with latest version of develop.

File: client/additional-methods-setup/upe-preview-methods-selector/test/enable-upe-preview-task.test.js
Comment (KarlisJ) [reply client/additional-methods-setup/upe-preview-methods-selector/test/enable-upe-preview-task.test.js:1]: No it tests functionality that is being removed from the codebase

File: changelog/add-6034-feature-flag-bnlp (hunk 4 above)
Comment (KarlisJ) [reply changelog/add-6034-feature-flag-bnlp]: > Maybe the latest changes in develop need to be merged in base branch?

It's already done. You can see [here is the file in the base branch](https://github.com/Automattic/woocommerce-payments/blob/develop-bnpl/changelog/add-6034-feature-flag-bnlp)

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `6417` or branch name `update/6274-add-bnpl-pm-to-additional-pm-screen` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=update/6274-add-bnpl-pm-to-additional-pm-screen) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 7e014ba681977dff88a61de4d09b23beaee7732b
- Build time: 2023-06-06 10:14:04 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Comment by KarlisJ** [comment]
@mgascam I've done the rebase as we discussed and tested that there are no braking changes if the server returns `false` for feature flag. 

Can you approve so we can merge and move forward with rest of PRs

**Review by mgascam — APPROVED** [review]
Looks good to me after the rebase onto `bnpl-develop`.

**Pull request merged**